### DISCUSSION TITLE
⛔ Re-enable Font Size Option for Heading Widget

### DISCUSSION THREAD

**Discussion by projectpi-ca** [elementor/elementor#25174] [Feature Request]
### Prerequisites

- [X] I have searched for similar feature requests in both open and closed discussions and cannot find an existing request.
- [X] I have verified that the feature is still missing from the latest stable versions of Elementor or Elementor Pro.

### What problem is your feature request going to solve? Please describe..

I managed over 20 sites that use Elementor. After updating to the latest version, the heading widget no longer shows the font size option.

I have a custom Hello child theme that utilizes this feature and allows the font sizes to be controlled globally. 

After looking into the code, I found the following lines in this file: elementor/includes/widgets/heading.php

```
		$this->add_control(
			'size',
			[
				'label' => esc_html__( 'Size', 'elementor' ),
				'type' => Controls_Manager::SELECT,
				'options' => [
					'default' => esc_html__( 'Default', 'elementor' ),
					'small' => esc_html__( 'Small', 'elementor' ),
					'medium' => esc_html__( 'Medium', 'elementor' ),
					'large' => esc_html__( 'Large', 'elementor' ),
					'xl' => esc_html__( 'XL', 'elementor' ),
					'xxl' => esc_html__( 'XXL', 'elementor' ),
				],
				'default' => 'default',
				'condition' => [
					'size!' => 'default', // a workaround to hide the control, unless it's in use (not default).
				],
			]
		);
```

It appears that the setting is disabled unless the size setting is already in use.

I reached out to Elementor support. They couldn't tell me how to re-enable it, or provide an explanation, but they told me to come here. I searched the GitHub and briefly checked the roadmap but couldn't find anything related to this. 

Can we have this feature back? If not, what's the reason for this and what's the plan for existing widgets that already utilize this feature? I can override this with custom code but an explanation would be appreciated.

### Describe the solution you'd like.

Bring back the feature.

### Describe alternatives you've considered.

Override it with custom code.

### Agreement

- [X] I agree that my request may be closed without action if it doesn't meet all the requirements.

**Comment by nicholaszein** [comment]
Hello @projectpi-ca,

Thank you for submitting a feature request! 🙏 

### ⛔ 🚀 After reviewing your request, we have decided not to add this to our roadmap.

This feature was **deprecated** and discontinued. This is in preparation for (and to avoid conflicts with) future enhancements in the Editor based on CSS Classes.

Existing widgets that have the size selected will remain as they are unless you edit them. If you wish to change the Typography size of Heading widgets you should go to the **`Style`** tab, and under **`Typography`** change the size there.

Please feel free to contact us again with other feature suggestions. 😉 

Best regards
